src/model/include/iDynTree/Model/Model.h

```diff
@@ -417,6 +417,18 @@ namespace iDynTree
          */
         LinkIndex getFrameLink(const FrameIndex frameIndex) const;
 
+        /**
+         * Get the additional frames of a specified link.
+         *
+         * @note The vector of returned frame index is ordered according to the frame index.
+         * @warning This method searches linearly over all the frames.
+         *
+         * @param[in] link a LinkIndex of the specified link,
+         * @param[out] frames a vector of FrameIndex of the frame indeces attached to the specified link index,
+         * @return true if the specified link is a valid link, false otherwise.
+         */
+        bool getLinkAdditionalFrames(const LinkIndex lnkIndex, std::vector<FrameIndex>& frameIndeces) const;
```

Comment:
Does it also make sense to have an overload with `linkName` as input argument?

Reply:
Similar overload for `frameNames` as output?

Reply:
It could make sense, but I would leave this to a separate PR. Probably we can have a guideline to have for all methods in `iDynTree::Model` both the index and the names one, and then have a PR for add names-variant for all methods missing them. Can you open a new issue on this?